server/src/main/java/org/opensearch/search/aggregations/AggregationCollectorManager.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.aggregations;
+
+import org.apache.lucene.search.Collector;
+import org.apache.lucene.search.CollectorManager;
+import org.opensearch.common.CheckedFunction;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.profile.query.InternalProfileCollector;
+import org.opensearch.search.query.ReduceableSearchResult;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Common {@link CollectorManager} used by both concurrent and non-concurrent aggregation path and also for global and non-global
+ * aggregation operators
+ */
+public class AggregationCollectorManager implements CollectorManager<Collector, ReduceableSearchResult> {
```

Comment:
Since this collector is not used as-is, let make it package private (non public)